src/main/flight/failsafe.c

```diff
@@ -164,26 +165,31 @@ void failsafeOnValidDataReceived(void)
     failsafeState.validRxDataReceivedAt = millis();
     if ((cmp32(failsafeState.validRxDataReceivedAt, failsafeState.validRxDataFailedAt) > (int32_t)failsafeState.rxDataRecoveryPeriod) ||
         (cmp32(failsafeState.validRxDataFailedAt, failsafeState.validRxDataReceivedAt) > 0)) {
+        // rxDataRecoveryPeriod is 1.2s
         failsafeState.rxLinkState = FAILSAFE_RXLINK_UP;
+        // Allow arming now we have an RX link
         unsetArmingDisabled(ARMING_DISABLED_RX_FAILSAFE);
     }
 }
 
 void failsafeOnValidDataFailed(void)
 {
-    setArmingDisabled(ARMING_DISABLED_RX_FAILSAFE); // To prevent arming with no RX link
     failsafeState.validRxDataFailedAt = millis();
-    if ((cmp32(failsafeState.validRxDataFailedAt, failsafeState.validRxDataReceivedAt) > (int32_t)failsafeState.rxDataFailurePeriod) ||
-        (IS_RC_MODE_ACTIVE(BOXFAILSAFE) && (failsafeConfig()->failsafe_switch_mode == FAILSAFE_SWITCH_MODE_STAGE2))) {
+    if ((cmp32(failsafeState.validRxDataFailedAt, failsafeState.validRxDataReceivedAt) > (int32_t)failsafeState.rxDataFailurePeriod)) {
+        // rxDataFailurePeriod is stage 1 guard time
         failsafeState.rxLinkState = FAILSAFE_RXLINK_DOWN;
+        // Prevent arming with no RX link
+        setArmingDisabled(ARMING_DISABLED_RX_FAILSAFE);
```

Comment:
Previously we experienced occasional, random, inability to arm during normal power-ups.  
`failsafeOnValidDataFailed` is now active immediately is a transient loss of link, so that the start of the timer to enter Stage2 is at the beginning of a loss of the link.
By moving the arming block to after Stage1 is complete, ie, when full Failsafe is declared, arming with a link has occasional packet dropouts is more likely to be successful.

Reply:
Please see below.